vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
@stevvooe Yes, since the type here is boolean the `oldValue` argument is optional. If you think this is better for readability, ie: just calling it `enableService` with one argument, I can change it. But what it is doing is a CAS operation.

Since its a libnetwork change I will modify it in the libnetwork PR.